content/bigquery.textile

```diff
@@ -0,0 +1,109 @@
+| *Partitioning* | _(Optional)_ The table must be created with the desired partitioning settings in BigQuery before making the rule in Ably. |
+| *Advanced settings* | Any additional configuration or custom fields relevant to your BigQuery setup (for future enhancements). |
+
+h4(#dashboard). Create a BigQuery rule in the dashboard
+
+The following steps to create a BigQuery rule using the Ably dashboard:
+
+* Log in to the "Ably dashboard":https://ably.com/accounts/any and select the application you want to stream data from.
+* Navigate to the *Integrations* tab.
+* Click *New integration rule*.
+* Select *Firehose*.
+* Choose *BigQuery* from the list of available Firehose integrations.
+* Configure the rule settings as described below.Then, click *Create*.
+
+h4(#api-rule). Create a BigQuery rule using the Control API
+
+The following steps to create a BigQuery rule using the "Control API:":https://ably.com/docs/api#control-api
+
+* Using the required "rules":/control-api#examples-rules to specify the following parameters:
+** @ruleType@: Set this to "bigquery" to define the rule as a BigQuery integration.
+** destinationTable: Specify the BigQuery table where the data will be stored.
+** @serviceAccountCredentials@: Provide the necessary GCP service account JSON key to authenticate and authorize data insertion.
+** @channelFilter@ (optional): Use a regular expression to apply the rule to specific channels.
+** @format@ (optional): Define the data format based on how you want messages to be structured.
+* Make an HTTP request to the Control API to create the rule.
+
+h3(#schema). JSON Schema
+
+You can run queries directly against the Ably-managed BigQuery table. For example, if the message payloads are stored as raw JSON in the data column, you can parse them using the following query:
+
+```[json]
+{
+“name”: “id”,
+“type”: “STRING”,
+“mode”: “REQUIRED”,
+“description”: “Unique ID assigned by Ably to this message. Can optionally be assigned by the client.”
+}
+```
+
+h3(#queries). Direct queries
+
+Run queries directly against the Ably-managed table. For instance, to parse JSON payloads stored in @data@:
+
+```[sql]
+SELECT
+PARSE_JSON(CAST(data AS STRING)) AS parsed_payload
+FROM project_id.dataset_id.table_id
+WHERE channel = “my-channel”
+```
+
+The following explains the components of the query:
+
+|_. Query Function |_. Purpose |
+| @CAST(data AS STRING)@ | Converts the data column from BYTES (if applicable) into a STRING format. |
+| @PARSE_JSON(…)@ | Parses the string into a structured JSON object for easier querying. |
+| @WHERE channel = “my-channel”@ | Filters results to retrieve messages only from a specific Ably channel. |
+
+<aside data-type='note'>
+<p>Parsing JSON at query time can be computationally expensive for large datasets. If your queries need frequent JSON parsing, consider pre-processing and storing structured fields in a secondary table using an ETL pipeline for better performance.</p>
+</aside>
```

Comment:
I think you could make this part of the ETL section below since it's a sub-section rather than calling it out as a note.